src/Microsoft.Extensions.Configuration.AzureAppConfiguration/OfflineFileCache.cs

```diff
@@ -308,5 +248,22 @@ internal static void ValidateCachePath(string path)
                 throw new ArgumentException($"The directory with path {directoryPath} does not exist.");
             }
         }
+
+        internal static void ValidateFileCacheExpiration(TimeSpan expiration)
+        {
+            if (expiration <= TimeSpan.Zero)
+            {
+                throw new ArgumentException($"Please provide a valid {nameof(OfflineFileCacheOptions)}.{nameof(OfflineFileCacheOptions.FileCacheExpiration)}.");
+            }
+
+            try
+            {
+                _ = DateTimeOffset.UtcNow + expiration;
+            }
+            catch (ArgumentOutOfRangeException)
+            {
+                throw new ArgumentOutOfRangeException($"{nameof(OfflineFileCacheOptions)}.{nameof(OfflineFileCacheOptions.FileCacheExpiration)}", "Please provide a shorter file cache expiration. The expiration time added to the current UTC time results in an un-representable DateTime.");

```

Comment:
> The expiration time added to the current UTC time results in an un-representable DateTime.

That's going pretty deep into the details. would probably be fine to just call out it's too big.